tools/spec-gen-sdk/src/automation/logging.ts

```diff
@@ -139,8 +139,8 @@ export const loggerWaitToFinish = async (logger: winston.Logger) => {
   for (const transport of logger.transports) {
     if (transport instanceof winston.transports.File) {
       if (transport.end) {
-          transport.end();
           await setTimeout(2000);
+          transport.end();
```

Comment:
Will 2000 seconds be not enough time if there are many logs?

Reply:
Based on the observation that the `NodeError: write after end` does not occur with small data sets, I analyzed the write times for each log. Regardless of the data size, the write time for each log is consistently between 1ms and 10ms. The log length itself does not cause this issue. The error arises because there are unwritten logs in the stream used by Winston. To address this, ensure a delay of a few milliseconds to write the queued logs. I believe the missing logs do not exceed 10, so a delay of **1000ms** or **2000ms** should suffice.

| Message Length | Cost Time |
| --- | --- |
| 30 | 5.408 ms |
| 196 | 4.408 ms |
| 52 | 0.590 ms |
| 149 | 3.554 ms |
| 100 | 3.554 ms |
| 28039 | 3.554 ms |